### PR TITLE
Allow providing a firefox binary through new session parameters.

### PR DESCRIPTION
**Pull request by jgraham**
Binary can be specified with firefox_binary capability. If none
is provided it falls back to the value provided with -b on the
command line, if any, and returns an error otherwise.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/geckodriver/66)

### CODE REVIEW

File: src/marionette.rs [hunk 1]
```diff
@@ -269,7 +273,26 @@ impl MarionetteHandler {
...
+                                        .ok_or(WebDriverError::new(
+                                            ErrorStatus::InvalidArgument,
+                                            "'firefox_binary' capability was not a string")))))
+            } else {
+                self.binary.as_ref().map(|x| x.clone())
+            };
+            if let Some(path) = binary {
+                BrowserLauncher::BinaryLauncher(path)
+            } else {
+                return Err(WebDriverError::new(ErrorStatus::UnknownError,
```
Comment (AutomatedTester) [inline src/marionette.rs]: Could we just try use the default install paths for each OS here instead of erroring?

Comment (andreastt) [inline src/marionette.rs]: Specifically https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/firefox/binary.rb#L135 describes a somewhat sane implementation.

**Comment by jgraham** [comment]
Can we land this without the default paths stuff, and add that in a second commit? This just required a painful rebase, and adding that isn't going to be tricky (it's like one extra clause in an if condition)

**Comment by andreastt** [comment]
That makes sense to me.

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/geckodriver/66#-:-KLHX93PmuneANpFcMpX:-351867022)*

**Pull request merged**